src/jltypes.c

```diff
@@ -2383,6 +2384,11 @@ void jl_init_types(void) JL_GC_DISABLED
                                        jl_perm_symsvec(2, "typ", "fields"),
                                        jl_svec2(jl_any_type, jl_array_any_type), 0, 0, 2);
 
+    jl_method_match_type = jl_new_datatype(jl_symbol("MethodMatch"), core, jl_any_type, jl_emptysvec,
+                                       jl_perm_symsvec(4, "metharg", "methsp", "method", "fully_covers"),
```

Comment:
Can we give these real words? (or at least the same mangling we do now):
`signature` (or `specTypes`), `env` (or `sparam_vals`), `method`, `<:` (or whatever this is)?

Reply:
This is what they were called in inlining. I agree we can pick better names. How about just `specTypes`, `sparams`, `method`, `fully_covers`? I don't like using `<:` as a field name.

Reply:
How about not mixing snake case and camel case? 😬

Reply:
`spec_types` then, though at some point we should also switch over `MethodInstance` then.